samza-api/src/main/java/org/apache/samza/operators/MessageStream.java

```diff
@@ -213,13 +213,13 @@
 
   /**
    * Re-partitions this {@link MessageStream} using keys from the {@code keyExtractor} by creating a new
-   * intermediate stream on the {@code job.default.system}. This intermediate stream is both an output and
-   * input to the job.
+   * intermediate stream on the default system provided via {@link StreamGraph#setDefaultSystem}. This intermediate
+   * stream is both an output and input to the job.
    * <p>
    * Uses the provided {@link KVSerde} for serialization of keys and values. If the provided {@code serde} is null,
-   * uses the default serde provided via {@link StreamGraph#setDefaultSerde}, which must be a KVSerde. If the default
-   * serde is not a {@link KVSerde}, a runtime exception will be thrown. If no default serde has been provided
-   * <b>before</b> calling this method, a {@code KVSerde<NoOpSerde, NoOpSerde>} is used.
+   * uses the default system's serde provided via {@link StreamGraph#setDefaultSystem}, which must be a KVSerde. If
+   * the default system's serde is not a {@link KVSerde}, a runtime exception will be thrown. If no default system
+   * has been set <b>before</b> calling this method, serde for the default system must be set in configuration.
```

Comment:
Since we don't support system level serdes, even if a default system is set using a SystemDescriptor, its serde must still be specified using configuration.